crates/ra_hir/src/ty.rs

```diff
@@ -526,6 +526,21 @@ struct InferenceContext<'a, D: HirDatabase> {
     return_ty: Ty,
 }
 
+// helper function that determines whether a binary operator
+// always returns a boolean
+fn is_boolean_operator(op: BinOp) -> bool {
+    match op {
+        BinOp::BooleanOr
+        | BinOp::BooleanAnd
+        | BinOp::EqualityTest
+        | BinOp::LesserEqualTest
+        | BinOp::GreaterEqualTest
+        | BinOp::LesserTest
+        | BinOp::GreaterTest => true,
+        _ => false,
```

Comment:
You've got an unreachable pattern warning failing the build here, I think it's fine to remove this line for now:
```suggestion
```

Reply:
Hmmm, I wonder why I didn't catch this locally. I cargo checked and tested the entire repo I think :thinking:

Reply:
Yeah, I think warnings are the only problem that will make the travis build fail that you don't catch with `cargo test`...